site-content/source/modules/ROOT/pages/development/ide.adoc

```diff
@@ -2,68 +2,84 @@
+Building Cassandra from source is the first important step in contributing to the Apache Cassandra project.
+You will need to install Java (JDK 8 or 11, depending on which version
+you want to build Cassandra against, best is to install both and then
+switch between them when needed), Git, and Ant (including ant-optional).
+
+=== Getting the source code
 
 The source code for Cassandra is shared on the central Apache Git
 repository and organized by branch, one branch for each major version.
-You can access the code for the current development branch using:
+You can access the code for the current development branch at
+https://gitbox.apache.org/repos/asf/cassandra.git (the primary location)
+or at https://github.com/apache/cassandra (a mirror location).
 
-// 01.15.21 - polandll - should this step be a fork rather than a clone?
+However, to develop a patch or a new feature, you should fork the Cassandra
+project first and then clone it from your own fork:
 
-[source, plaintext]
+[source,plaintext]
 ----
-git clone https://gitbox.apache.org/repos/asf/cassandra.git cassandra-trunk
+git clone https://github.com/<your_git_name>/cassandra.git cassandra
 ----
 
-Other branches will point to different versions of Cassandra. Switching
-to a different branch requires checking out the branch.
-For example, to checkout the latest version of Cassandra 3.0, use:
+=== Building Cassandra
 
-[source, plaintext]
-----
-git checkout cassandra-3.0
-----
+Cassandra is a Java project which is built using Ant.
+The build file, `build.xml` located in the root of the project content,
+has various tasks defined (you can list all of them with some short
+descriptions by running `ant -p`).
 
-You can get a list of available branches with `git branch`.
+The build uses the Java compiler which is currently set up in your
+shell. By default, the build uses Java 8. If you want to build
+with Java 11, you need to either add a build property `-Duse.jdk11=true`
+to your Ant build command or export the environment variable
+`CASSANDRA_USE_JDK11=true`. Otherwise, if you run the build using
+Java 11, the build script complains.
 
-Build Cassandra using ant:
+Now you can build Cassandra with the default task - just execute
+`ant` or `ant jar`. This may take a significant amount of time depending
+on artifacts that have to be downloaded or the number of classes that
+need to be compiled. When the build completes, you can find a JAR file
+in the build directory, and the database service can be started with
+the `bin/cassandra` script.
 
-[source, plaintext]
-----
-ant
-----
+=== Building a distribution
 
-This may take a significant amount of time depending on artifacts that have to
-be downloaded or the number of classes that need to be compiled.
+Some tasks you might be interested in are:
+
+* `sources-jar` - produce accompanying JAR for the primary Cassandra
+artifact (also in the build directory)
```

Comment:
Not sure who uses `sources-jar`. Mostly an artefact for releases.

I would replace this with the `build` and `jar`. (It's ok to mention again `jar`, and state here that it is the default.)

Reply:
I put there `source-jar` and `javadoc-jar` for completeness, but I'm fine with removing them. I'll leave `build`, `build-test`, `artifacts`, `mvn-install` and `realclean` as they seem to be more useful